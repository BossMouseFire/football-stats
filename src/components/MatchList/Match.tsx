import React, { useMemo } from 'react';
import { labelsMatch, MatchProps } from './MatchListProps';
import { getDate } from '../../utils/date';
import styles from './MatchList.module.scss';
import clsx from 'clsx';
import { Winner } from '../../dto/ScoreDto';

export const Match: React.FC<MatchProps> = ({ match, className }) => {
    const dateMatch = useMemo(() => {
        return getDate(match.utcDate);
    }, [match.utcDate]);

    const isWinnerHomeTeam = !!(
        match.score.winner && match.score.winner === Winner.HOME_TEAM
    );

    const isWinnerAwayTeam = !!(
        match.score.winner && match.score.winner === Winner.AWAY_TEAM
    );

    return (
        <div className={clsx(styles.match, className)}>
            <div className={styles.status}>{labelsMatch[match.status]}</div>
            <div className={styles.date}>{dateMatch}</div>
            <div
                className={clsx(
                    styles.team,
                    isWinnerHomeTeam && styles.winnerTeam,
                )}
            >
                <span>{match.homeTeam.name}</span>
                <img src={match.homeTeam.crest} alt={match.homeTeam.name} />
            </div>
            <div className={styles.score}>
                <div>
                    {match.score.winner ? match.score.fullTime.home : '-'}
                </div>
                :
                <div>
                    {match.score.winner ? match.score.fullTime.away : '-'}
                </div>
            </div>
            <div
                className={clsx(
                    styles.team,
                    isWinnerAwayTeam && styles.winnerTeam,
                )}
            >
                <span>{match.awayTeam.name}</span>
                <img src={match.awayTeam.crest} alt={match.awayTeam.name} />
            </div>
        </div>
    );
};
