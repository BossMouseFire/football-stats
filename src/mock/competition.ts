export const competitionMock = {
    area: {
        id: 2224,
        name: 'Spain',
        code: 'ESP',
        flag: 'https://crests.football-data.org/760.svg',
    },
    id: 2014,
    name: 'Primera Division',
    code: 'PD',
    type: 'LEAGUE',
    emblem: 'https://crests.football-data.org/PD.png',
    currentSeason: {
        id: 1577,
        startDate: '2023-08-13',
        endDate: '2024-05-26',
        currentMatchday: 13,
        winner: null,
    },
    lastUpdated: '2022-03-20T09:20:08Z',
};

export const competitionMockWithWinner = {
    area: {
        id: 2267,
        name: 'World',
        code: 'INT',
        flag: null,
    },
    id: 2000,
    name: 'FIFA World Cup',
    code: 'WC',
    type: 'CUP',
    emblem: 'https://crests.football-data.org/qatar.png',
    currentSeason: {
        id: 1382,
        startDate: '2022-11-20',
        endDate: '2022-12-18',
        currentMatchday: 8,
        winner: {
            id: 762,
            name: 'Argentina',
            shortName: 'Argentina',
            tla: 'ARG',
            crest: 'https://crests.football-data.org/762.png',
            address: 'Viamonte 1366/76 Buenos Aires, Buenos Aires 1053',
            website: 'http://www.afa.org.ar',
            founded: 1893,
            clubColors: 'Sky Blue / White / Black',
            venue: null,
            lastUpdated: '2022-05-17T21:09:25Z',
        },
    },
    lastUpdated: '2022-05-09T19:45:29Z',
};
