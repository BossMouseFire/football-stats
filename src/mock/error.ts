import { ErrorData } from '../errors/ErrorData';

export const errorMock404 = {
    message: 'The resource you are looking for does not exist.',
    error: 404,
};

export const errorMock400 = new ErrorData({
    message:
        "Argument 'competitionId' is expected to be either an integer in a specified range or a competition code.",
    errorCode: 400,
});

export const errorMock403 = new ErrorData({
    message:
        'The resource you are looking for is restricted and apparently not within your permissions. Please check your subscription.',
    errorCode: 403,
});
