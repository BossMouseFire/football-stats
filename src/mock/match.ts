import { MatchDto } from '../dto/MatchDto';

export const matchesMock: MatchDto[] = [
    {
        season: {
            id: 1577,
            startDate: '2023-08-13',
            endDate: '2024-05-26',
            currentMatchday: 13,
            winner: null,
        },
        id: 438482,
        utcDate: '2023-08-11T17:30:00Z',
        status: 'FINISHED',
        homeTeam: {
            id: 267,
            name: 'UD Almería',
            shortName: 'Almería',
            tla: 'ALM',
            crest: 'https://crests.football-data.org/267.png',
        },
        awayTeam: {
            id: 87,
            name: 'Rayo Vallecano de Madrid',
            shortName: 'Rayo Vallecano',
            tla: 'RAY',
            crest: 'https://crests.football-data.org/87.svg',
        },
        score: {
            winner: 'AWAY_TEAM',
            duration: 'REGULAR',
            fullTime: {
                home: 0,
                away: 2,
            },
            halfTime: {
                home: 0,
                away: 2,
            },
        },
    },
    {
        season: {
            id: 1577,
            startDate: '2023-08-13',
            endDate: '2024-05-26',
            currentMatchday: 13,
            winner: null,
        },
        id: 438479,
        utcDate: '2023-08-11T20:00:00Z',
        status: 'FINISHED',
        homeTeam: {
            id: 559,
            name: 'Sevilla FC',
            shortName: 'Sevilla FC',
            tla: 'SEV',
            crest: 'https://crests.football-data.org/559.svg',
        },
        awayTeam: {
            id: 95,
            name: 'Valencia CF',
            shortName: 'Valencia',
            tla: 'VAL',
            crest: 'https://crests.football-data.org/95.svg',
        },
        score: {
            winner: 'AWAY_TEAM',
            duration: 'REGULAR',
            fullTime: {
                home: 1,
                away: 2,
            },
            halfTime: {
                home: 0,
                away: 0,
            },
        },
    },
];
