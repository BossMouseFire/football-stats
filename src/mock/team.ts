import { TeamDto } from '../dto/TeamDto';

export const teamsMock: TeamDto[] = [
    {
        id: 511,
        name: 'Toulouse FC',
        shortName: 'Toulouse',
        tla: 'TOU',
        crest: 'https://crests.football-data.org/511.png',
        address: '1 allée Gabriel Biénèsm, BP 54023 Toulouse 31028',
        website: 'http://www.tfc.info',
        founded: 1937,
        clubColors: 'Violet / White',
        venue: 'Stadium Municipal',
    },
];
