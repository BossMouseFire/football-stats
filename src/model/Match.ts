import { TableTeam } from './TableTeam';
import { MatchDto } from '../dto/MatchDto';
import { Season } from './Season';

export class Match {
    id: number;
    status: string;
    utcDate: string;
    season: Season;
    score: Season;
    homeTeam: TableTeam;
    awayTeam: TableTeam;

    constructor(dto: MatchDto) {
        this.id = dto.id;
        this.status = dto.status;
        this.utcDate = dto.utcDate;
        this.season = new Season(dto.season);
        this.score = new Season(dto.score);
        this.homeTeam = new TableTeam(dto.homeTeam);
        this.awayTeam = new TableTeam(dto.awayTeam);
    }
}
