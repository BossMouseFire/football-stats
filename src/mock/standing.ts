import { Standing } from '../model/Standing';
import { Competition } from '../model/Competition';
import { StandingDto } from '../dto/StandingDto';
import { CompetitionDto } from '../dto/CompetitionDto';
import { SeasonDto } from '../dto/SeasonDto';

type StandingCompetition = Pick<
    CompetitionDto,
    'id' | 'name' | 'emblem' | 'area' | 'code' | 'type'
>;

export const standingsMock: {
    standings: StandingDto[];
    competition: StandingCompetition;
    season: SeasonDto;
} = {
    competition: {
        id: 2015,
        name: 'Ligue 1',
        code: 'FL1',
        type: 'LEAGUE',
        emblem: 'https://crests.football-data.org/FL1.png',
    },
    season: {
        id: 1595,
        startDate: '2023-08-13',
        endDate: '2024-05-18',
        currentMatchday: 12,
        winner: null,
    },
    standings: [
        {
            stage: 'REGULAR_SEASON',
            type: 'TOTAL',
            table: [
                {
                    position: 1,
                    team: {
                        id: 524,
                        name: 'Paris Saint-Germain FC',
                        shortName: 'PSG',
                        tla: 'PSG',
                        crest: 'https://crests.football-data.org/524.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 8,
                    draw: 3,
                    lost: 1,
                    points: 27,
                    goalsFor: 29,
                    goalsAgainst: 9,
                    goalDifference: 20,
                },
                {
                    position: 2,
                    team: {
                        id: 522,
                        name: 'OGC Nice',
                        shortName: 'Nice',
                        tla: 'NIC',
                        crest: 'https://crests.football-data.org/522.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 7,
                    draw: 5,
                    lost: 0,
                    points: 26,
                    goalsFor: 13,
                    goalsAgainst: 4,
                    goalDifference: 9,
                },
                {
                    position: 3,
                    team: {
                        id: 548,
                        name: 'AS Monaco FC',
                        shortName: 'Monaco',
                        tla: 'ASM',
                        crest: 'https://crests.football-data.org/548.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 7,
                    draw: 3,
                    lost: 2,
                    points: 24,
                    goalsFor: 25,
                    goalsAgainst: 14,
                    goalDifference: 11,
                },
                {
                    position: 4,
                    team: {
                        id: 521,
                        name: 'Lille OSC',
                        shortName: 'Lille',
                        tla: 'LIL',
                        crest: 'https://crests.football-data.org/521.svg',
                    },
                    playedGames: 12,
                    form: null,
                    won: 5,
                    draw: 5,
                    lost: 2,
                    points: 20,
                    goalsFor: 15,
                    goalsAgainst: 11,
                    goalDifference: 4,
                },
                {
                    position: 5,
                    team: {
                        id: 547,
                        name: 'Stade de Reims',
                        shortName: 'Stade de Reims',
                        tla: 'SDR',
                        crest: 'https://crests.football-data.org/547.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 6,
                    draw: 2,
                    lost: 4,
                    points: 20,
                    goalsFor: 17,
                    goalsAgainst: 15,
                    goalDifference: 2,
                },
                {
                    position: 6,
                    team: {
                        id: 533,
                        name: 'Le Havre AC',
                        shortName: 'Le Havre',
                        tla: 'HAC',
                        crest: 'https://crests.football-data.org/533.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 3,
                    draw: 6,
                    lost: 3,
                    points: 15,
                    goalsFor: 12,
                    goalsAgainst: 13,
                    goalDifference: -1,
                },
                {
                    position: 7,
                    team: {
                        id: 512,
                        name: 'Stade Brestois 29',
                        shortName: 'Brest',
                        tla: 'BRE',
                        crest: 'https://crests.football-data.org/512.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 4,
                    draw: 3,
                    lost: 4,
                    points: 15,
                    goalsFor: 11,
                    goalsAgainst: 13,
                    goalDifference: -2,
                },
                {
                    position: 8,
                    team: {
                        id: 543,
                        name: 'FC Nantes',
                        shortName: 'Nantes',
                        tla: 'NAN',
                        crest: 'https://crests.football-data.org/543.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 4,
                    draw: 2,
                    lost: 6,
                    points: 14,
                    goalsFor: 17,
                    goalsAgainst: 23,
                    goalDifference: -6,
                },
                {
                    position: 9,
                    team: {
                        id: 516,
                        name: 'Olympique de Marseille',
                        shortName: 'Marseille',
                        tla: 'MAR',
                        crest: 'https://crests.football-data.org/516.png',
                    },
                    playedGames: 10,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 3,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 12,
                    goalDifference: 0,
                },
                {
                    position: 10,
                    team: {
                        id: 546,
                        name: 'Racing Club de Lens',
                        shortName: 'RC Lens',
                        tla: 'RCL',
                        crest: 'https://crests.football-data.org/546.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 4,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 13,
                    goalDifference: -1,
                },
                {
                    position: 11,
                    team: {
                        id: 545,
                        name: 'FC Metz',
                        shortName: 'FC Metz',
                        tla: 'FCM',
                        crest: 'https://crests.football-data.org/545.svg',
                    },
                    playedGames: 12,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 5,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 18,
                    goalDifference: -6,
                },
                {
                    position: 12,
                    team: {
                        id: 518,
                        name: 'Montpellier HSC',
                        shortName: 'Montpellier',
                        tla: 'MON',
                        crest: 'https://crests.football-data.org/518.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 4,
                    points: 12,
                    goalsFor: 15,
                    goalsAgainst: 14,
                    goalDifference: 1,
                },
                {
                    position: 13,
                    team: {
                        id: 529,
                        name: 'Stade Rennais FC 1901',
                        shortName: 'Stade Rennais',
                        tla: 'REN',
                        crest: 'https://crests.football-data.org/529.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 6,
                    lost: 4,
                    points: 12,
                    goalsFor: 16,
                    goalsAgainst: 16,
                    goalDifference: 0,
                },
                {
                    position: 14,
                    team: {
                        id: 511,
                        name: 'Toulouse FC',
                        shortName: 'Toulouse',
                        tla: 'TOU',
                        crest: 'https://crests.football-data.org/511.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 6,
                    lost: 4,
                    points: 12,
                    goalsFor: 13,
                    goalsAgainst: 16,
                    goalDifference: -3,
                },
                {
                    position: 15,
                    team: {
                        id: 576,
                        name: 'RC Strasbourg Alsace',
                        shortName: 'Strasbourg',
                        tla: 'RC ',
                        crest: 'https://crests.football-data.org/576.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 3,
                    lost: 5,
                    points: 12,
                    goalsFor: 9,
                    goalsAgainst: 15,
                    goalDifference: -6,
                },
                {
                    position: 16,
                    team: {
                        id: 525,
                        name: 'FC Lorient',
                        shortName: 'Lorient',
                        tla: 'FCL',
                        crest: 'https://crests.football-data.org/525.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 5,
                    lost: 5,
                    points: 11,
                    goalsFor: 15,
                    goalsAgainst: 21,
                    goalDifference: -6,
                },
                {
                    position: 17,
                    team: {
                        id: 541,
                        name: 'Clermont Foot 63',
                        shortName: 'Clermont Foot',
                        tla: 'CLF',
                        crest: 'https://crests.football-data.org/541.svg',
                    },
                    playedGames: 11,
                    form: null,
                    won: 2,
                    draw: 3,
                    lost: 6,
                    points: 9,
                    goalsFor: 8,
                    goalsAgainst: 14,
                    goalDifference: -6,
                },
                {
                    position: 18,
                    team: {
                        id: 523,
                        name: 'Olympique Lyonnais',
                        shortName: 'Olympique Lyon',
                        tla: 'LYO',
                        crest: 'https://crests.football-data.org/523.svg',
                    },
                    playedGames: 11,
                    form: null,
                    won: 1,
                    draw: 4,
                    lost: 6,
                    points: 7,
                    goalsFor: 9,
                    goalsAgainst: 19,
                    goalDifference: -10,
                },
            ],
        },
    ],
};

export const standingsMockRefactor = {
    standings: standingsMock.standings.map((item) => new Standing(item)),
    competition: new Competition({
        ...standingsMock.competition,
        currentSeason: standingsMock.season,
    }),
};
