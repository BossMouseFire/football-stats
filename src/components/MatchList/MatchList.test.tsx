import { Match } from './Match';
import { renderToJson } from '../../test/helpers/renderToJson';
import { matchesMock } from '../../mock/match';

it('Match renders correctly', () => {
    const table = renderToJson(<Match match={matchesMock[0]} />);

    expect(table).toMatchSnapshot();
});

export {};
