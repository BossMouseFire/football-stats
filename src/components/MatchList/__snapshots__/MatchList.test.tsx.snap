// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match renders correctly 1`] = `
<div
  className="match"
>
  <div
    className="status"
  >
    Завершено
  </div>
  <div
    className="date"
  >
    11.08.2023
  </div>
  <div
    className="team"
  >
    <span>
      UD Almería
    </span>
    <img
      alt="UD Almería"
      src="https://crests.football-data.org/267.png"
    />
  </div>
  <div
    className="score"
  >
    <div>
      0
    </div>
    :
    <div>
      2
    </div>
  </div>
  <div
    className="team winnerTeam"
  >
    <span>
      Rayo Vallecano de Madrid
    </span>
    <img
      alt="Rayo Vallecano de Madrid"
      src="https://crests.football-data.org/87.svg"
    />
  </div>
</div>
`;
